more_itertools/more.py

```diff
@@ -2419,43 +2418,39 @@ def set_partitions(iterable, k=None):
     ...     print([''.join(p) for p in part])
     ['abc']
     ['a', 'bc']
+    ['ab', 'c']
     ['b', 'ac']
-    ['c', 'ab']
     ['a', 'b', 'c']
 
     """
-    iterable = tuple(iterable)
-    n = len(iterable)
-
-    def less(a, b):
-        """Orders index tuples lexically"""
-        return (a < b) if (len(a) == len(b)) else (len(a) < len(b))
-
-    def part_inds(inds, k, prev):
-        """Generates set partitions by index"""
-        if (k <= 1) and (less(prev, inds)):
-            yield (inds,)
+    L = list(iterable)
+    n = len(L)
+    if k is not None:
+        if k < 1:
+            raise ValueError(
```

Comment:
[Coveralls](https://coveralls.io/builds/24840087/source?filename=more_itertools%2Fmore.py#L2430) says this branch is not tested - could you test it?